src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -348,7 +348,10 @@ export class SavedObjectsClient {
       workspaces: 'workspaces',
     };
 
-    const renamedQuery = renameKeys<SavedObjectsFindOptions, any>(renameMap, options);
+    const renamedQuery = renameKeys<Omit<SavedObjectsFindOptions, 'ACLSearchParams'>, any>(
```

Comment:
But it shouldn't allow to pass ACLSearchParams from UI, am I right? This SavedObjectsClient is in `public` and it's supposed to be used from UI if I'm not mistaken.

So it looks like we should exclude it from the type definition `SavedObjectsFindOptions` so that the `find` function doesn't accept it through `options`:
```
  public find = async <T = unknown>(
    options: SavedObjectsFindOptions
  ) => {}
```

Reply:
I'm OK to omit `ACLSearchParams` from public `SavedObjectsFindOptions`. For now we won't call public SavedObjectsClient with ACLSearchParams from the UI. We will call workspace list API with permission modes, and then pass ACLSearchParams to server SavedObjectsClient.

Reply:
Updated.

Reply:
Addressed in https://github.com/wanglam/OpenSearch-Dashboards/pull/4/files#diff-4cb0b61333753382f1d44c371cd7d20d90bb6ce5f5aedbaca5b2a5b151f72af6